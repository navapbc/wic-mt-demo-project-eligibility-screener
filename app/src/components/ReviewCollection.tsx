import { UrlObject } from 'url'

import ButtonLink from '@components/ButtonLink'
import ReviewElement, { ReviewElementProps } from '@components/ReviewElement'
import TransLine from '@components/TransLine'

import { I18nKey } from '@src/types'

export type ReviewCollectionProps = {
  headerKey: I18nKey
  reviewElements: ReviewElementProps[]
  editable: boolean
  editHref: UrlObject | string
  firstElement?: boolean
}

const ReviewCollection = (props: ReviewCollectionProps) => {
  const {
    headerKey,
    reviewElements,
    editable = false,
    editHref = '',
    firstElement = false,
  } = props

  const marginTop = firstElement ? 'margin-top-3' : 'margin-top-6'

  return (
    <div className={`review-collection ${marginTop} border-bottom-1px`}>
      <h2>
        <TransLine i18nKey={headerKey} />
        {editable && (
          <div className="float-right">
            <ButtonLink labelKey="edit" href={editHref} style="unstyled" />
          </div>
        )}
      </h2>
      <dl className="margin-bottom-2">
        {reviewElements.map((element: ReviewElementProps, index: number) => (
          <ReviewElement key={index} {...element} />
        ))}
      </dl>
    </div>
  )
}

export default ReviewCollection
