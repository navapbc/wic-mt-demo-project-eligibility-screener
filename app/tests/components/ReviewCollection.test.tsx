import { render, screen } from '@testing-library/react'

import ReviewCollection from '@components/ReviewCollection'

const reviewElements = [
  { labelKey: 'element a', children: 'child a' },
  { labelKey: 'element b', children: 'child b' },
]

const testProps = {
  headerKey: 'header key',
  reviewElements: reviewElements,
  editable: false,
  editHref: '/edit-link',
  firstElement: false,
}

it('should match snapshot when it is the first element', () => {
  const { container } = render(
    <ReviewCollection {...testProps} firstElement={true} />
  )
  expect(container).toMatchSnapshot()
})

it('should match snapshot when it is not the first element', () => {
  const { container } = render(
    <ReviewCollection {...testProps} firstElement={false} />
  )
  expect(container).toMatchSnapshot()
})
