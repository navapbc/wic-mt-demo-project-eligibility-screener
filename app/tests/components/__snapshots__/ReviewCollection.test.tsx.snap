// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when it is not the first element 1`] = `
<div>
  <div
    class="review-collection margin-top-6 border-bottom-1px"
  >
    <h2>
      header key
    </h2>
    <dl
      class="margin-bottom-2"
    >
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            element a
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          child a
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            element b
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          child b
        </dd>
      </div>
    </dl>
  </div>
</div>
`;

exports[`should match snapshot when it is the first element 1`] = `
<div>
  <div
    class="review-collection margin-top-3 border-bottom-1px"
  >
    <h2>
      header key
    </h2>
    <dl
      class="margin-bottom-2"
    >
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            element a
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          child a
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            element b
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          child b
        </dd>
      </div>
    </dl>
  </div>
</div>
`;
